crates/host/src/wasmbus/mod.rs

```diff
@@ -3727,6 +3780,30 @@ fn ensure_actor_capability(
     Ok(())
 }
 
+fn injector_to_headers(injector: &OtelHeaderInjector) -> HeaderMap {
+    injector
+        .as_ref()
+        .iter()
+        .filter_map(|(k, v)| {
+            // There's not really anything we can do about headers that don't parse
+            let name = HeaderName::from_str(k.as_str()).ok();
+            let value = HeaderValue::from_str(v.as_str()).ok();
+            match (name, value) {
+                (Some(name), Some(value)) => Some((name, value)),
+                _ => None,
+            }
+        })
+        .collect()
+}
+
+fn headermap_to_headers(headermap: Option<HeaderMap>) -> HashMap<String, String> {
+    headermap
+        .unwrap_or_default()
+        .iter()
+        .map(|(k, v)| (k.to_string(), v.into()))
+        .collect()
+}
```

Comment:
It's weird for this function to take an `Option`, especially since `attach_span_context` does not do anything with an empty map, so IMO the caller should check if it's `None` and only call this when it's `Some`

Reply:
Oh actually then we bring back in a dependency on nats in the wasmcloud_tracing crate, which is what we've been trying to avoid